mettle/worker/worker.go

```diff
@@ -715,28 +715,32 @@ func (w *worker) execute(req *pb.ExecuteRequest, action *pb.Action, command *pb.
 }
 
 func containsAllOutputPaths(cmd *pb.Command, ar *pb.ActionResult) bool {
-	paths := make(map[string]struct{}, len(ar.OutputDirectories)+len(ar.OutputFiles)+len(ar.OutputSymlinks)+len(ar.OutputDirectorySymlinks))
+	paths := make(map[string]bool, len(ar.OutputDirectories)+len(ar.OutputFiles)+len(ar.OutputSymlinks)+len(ar.OutputDirectorySymlinks)+len(ar.OutputFileSymlinks))
 
 	// Build up the list of outputs that were generated
-	for _, f := range ar.OutputFiles {
-		paths[f.Path] = struct{}{}
+	for _, o := range ar.OutputFiles {
+		paths[o.Path] = true
 	}
 
-	for _, f := range ar.OutputDirectories {
-		paths[f.Path] = struct{}{}
+	for _, o := range ar.OutputDirectories {
+		paths[o.Path] = true
 	}
 
-	for _, f := range ar.OutputSymlinks {
-		paths[f.Path] = struct{}{}
+	for _, o := range ar.OutputSymlinks {
+		paths[o.Path] = true
 	}
 
-	for _, d := range ar.OutputDirectorySymlinks {
-		paths[d.Path] = struct{}{}
+	for _, o := range ar.OutputDirectorySymlinks {
+		paths[o.Path] = true
+	}
+
+	for _, o := range ar.OutputFileSymlinks {
```

Comment:
I guess it makes sense to check all these, although IIRC the two specific symlink types are obsolete now.

Reply:
Yeah I'm not sure about this either. We started seeing the bug at the same time this function was added. Both local and remote checks should be the same anyway so I thought it worth adding.